src/schedule/mod.rs

```diff
@@ -139,13 +149,26 @@ impl EpochSecrets {
             init_secret,
         }
     }
+
+    /// Remove the `encryption_secret` from the `EpochSecrets`, replacing it
+    /// with `None` and return it.
+    pub fn remove_encryption_secret(&self) -> Result<Secret, KeyScheduleError> {
+        // Remove the encryption secret by replacing it with `None`.
+        let encryption_secret = match self.encryption_secret.replace(None) {
+            Some(es) => es,
+            None => return Err(KeyScheduleError::SecretReuseError),
+        };
+        // Pass ownership to the `SecretTree` constructor, so that the
+        // `encryption_secret` is dropped afterwards.
+        Ok(encryption_secret)
+    }
 }
 
 impl Codec for EpochSecrets {
```

Comment:
Do we need the encoding at all btw? I think all we need is persistence going forward.

Reply:
Doesn't seem like it. I had already closed #207, but I just reopened it, so the issue is visible. I won't tackle it in this PR, though.